### PR TITLE
fixes a lot of stuff in NSIS

### PR DESCRIPTION
**Pull request by jpfeuffer**
## Description

- Removes badly working UAC code
- Uses Multiuser plugin to allow per user and all user install
- Speeds up installation significantly by putting SetOutputPath before the uninstall log macros
- Moving plugins into include folder
- Fixes stuff and better help message for command line/silent install
- Fixes Qt plugin installation
- TODO cleanup of unused code and fix uninstallation. Somehow it does not uninstall files. Only registry etc.

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

File: cmake/Windows/NSIS.template.in [hunk 1]
```diff
@@ -84,14 +83,15 @@ RequestExecutionLevel user    /* RequestExecutionLevel REQUIRED! */
 !include Sections.nsh
 !include Library.nsh
 !include FileFunc.nsh
-!include ${NSISRESOURCES}\UAC.nsh
-!include ${NSISRESOURCES}\IncludeScript_Misc.nsh
+#!include ${NSISRESOURCES}\Include\UAC.nsh
```
Comment (timosachsenberg) [inline cmake/Windows/NSIS.template.in:86]: do elevations still work without the UAC plugin? so it is not really required?
what about x64 and logiclib? not required anymore?

File: cmake/Windows/NSIS.template.in [hunk 2]
```diff
@@ -486,10 +528,16 @@ Section "-PathInst" SEC_PathRegister
...
       FindClose $0
 
-    #create OPENMS_DATA_PATH environment variable (for shared xml files etc)
-    ; set variable
-    WriteRegExpandStr ${env_hklm} "OPENMS_DATA_PATH" "$INSTDIR\share\OpenMS"
-    ; make sure windows knows about the change
+    # Create OPENMS_DATA_PATH environment variable (for shared xml files etc)
+
+    # Set variable
+    ${If} $MultiUser.InstallMode == "AllUsers"
```
Comment (timosachsenberg) [inline cmake/Windows/NSIS.template.in:534]: is this the If from logiclib or build in?

File: cmake/Windows/NSIS.template.in [hunk 3]
```diff
@@ -507,42 +555,61 @@ SectionGroup "Register File Extensions" SEC_RegisterExt
...
+#    SetOutPath $INSTDIR
+#    WriteUninstaller $INSTDIR\uninstall.exe
+#    WriteRegStr SHCTX "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\$(^Name)" DisplayName "$(^Name)"
+#    WriteRegStr SHCTX "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\$(^Name)" DisplayVersion "${VERSION}"
+#    WriteRegStr SHCTX "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\$(^Name)" Publisher "OpenMS Developer Team"
+#    WriteRegStr SHCTX "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\$(^Name)" URLInfoAbout "${URL}"
+#    WriteRegStr SHCTX "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\$(^Name)" DisplayIcon $INSTDIR\uninstall.exe
+#    WriteRegStr SHCTX "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\$(^Name)" UninstallString $INSTDIR\uninstall.exe
+#    WriteRegDWORD SHCTX "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\$(^Name)" NoModify 1
+#    WriteRegDWORD SHCTX "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\$(^Name)" NoRepair 1
```
Comment (timosachsenberg) [inline cmake/Windows/NSIS.template.in:602]: this part is now done by the plugin?

**Comment by timosachsenberg** [comment]
Wow. Looks good.
I wonder if we really can do without the UAC plugin though.

File: cmake/Windows/NSIS.template.in (hunk 1 above)
Comment (jpfeuffer) [reply cmake/Windows/NSIS.template.in:86]: They are in the Multiuser plugin.

File: cmake/Windows/NSIS.template.in (hunk 3 above)
Comment (jpfeuffer) [reply cmake/Windows/NSIS.template.in:602]: Yes

File: cmake/Windows/NSIS.template.in (hunk 2 above)
Comment (jpfeuffer) [reply cmake/Windows/NSIS.template.in:534]: I think logiclib. As mentioned they are included by the Multiuser plugin.

**Comment by poshul** [comment]
This sparks joy. I'm building it and will test it on windows.

**Pull request merged**